src/AWS.Deploy.CLI/Commands/GenerateDeploymentProjectCommand.cs

```diff
@@ -0,0 +1,218 @@
+        /// <returns>A list of <see cref="Recommendation"/></returns>
+        private async Task<List<Recommendation>> GenerateDeploymentRecommendations(Orchestrator orchestrator)
+        {
+            var recommendations = await orchestrator.GenerateDeploymentRecommendations(forDeployment: false);
+            if (recommendations.Count == 0)
+            {
+                throw new FailedToGenerateAnyRecommendations("The project you are trying to deploy is currently not supported.");
+            }
+            return recommendations;
+        }
+
+        /// <summary>
+        /// This method takes the path to the target deployment project and creates
+        /// a default save directory inside the parent folder of the current directory.
+        /// For example:
+        /// Target project directory - C:\Codebase\MyWebApp
+        /// Generated default save directory - C:\Codebase\MyWebApp.Deployment If the save directory already exists, then a suffix number is attached.
+        /// </summary>
+        /// <returns>The defaukt save directory path.</returns>
+        private string GenerateDefaultSaveDirectoryPath()
+        {
+            var applicatonDirectoryFullPath = _directoryManager.GetDirectoryInfo(_targetApplicationFullPath).Parent.FullName;
+            var saveCdkDirectoryFullPath = applicatonDirectoryFullPath + ".Deployment";
+
+            var suffixNumber = 0;
+            while (_directoryManager.Exists(saveCdkDirectoryFullPath))
+                saveCdkDirectoryFullPath = applicatonDirectoryFullPath + $".Deployment{++suffixNumber}";
+            
+            return saveCdkDirectoryFullPath;
+        }
+
+        /// <summary>
+        /// This method takes a path and creates a new directory at this path if one does not already exist.
+        /// </summary>
+        /// <param name="saveCdkDirectoryPath">Relative or absolute path of the directory at which the CDK deployment project will be saved.</param>
+        /// <returns>A boolean to indicate if a new directory was created.</returns>
+        private bool CreateSaveCdkDirectory(string saveCdkDirectoryPath)
+        {
+            var newDirectoryCreated = false;
+            if (!_directoryManager.Exists(saveCdkDirectoryPath))
+            {
+                _directoryManager.CreateDirectory(saveCdkDirectoryPath);
+                newDirectoryCreated = true;
+            }
+            return newDirectoryCreated;   
+        }
+
+        /// <summary>
+        /// This method takes the path to the intended location of the CDK deployment project and performs validations on it.
+        /// </summary>
+        /// <param name="saveCdkDirectoryPath">Relative or absolute path of the directory at which the CDK deployment project will be saved.</param>
+        /// <returns>A validation error message or an empty string if there are no errors.</returns>
+        private string ValidateSaveCdkDirectory(string saveCdkDirectoryPath)
```

Comment:
We shouldn't rely on the error message string for the state. I expect this method to return a boolean.

If you need message, you can also return a boolean and string tuple.

Reply:
Fixed

Reply:
Fixed